contract/src/interfaces/Types.sol

```diff
@@ -11,39 +11,53 @@ struct Signature {
     bytes32 s;
 }
 
+struct Amount {
+  address token; // ERC-20 token contract address
+  uint256 amount; // Token amount allocated
+}
+
 struct Allocation {
     address destination; // Where funds are sent on channel closure
     address token; // ERC-20 token contract address
     uint256 amount; // Token amount allocated
+    //TODO: consider using Amount
```

Comment:
Nested structs CAN be hard to handle in TS and Go (if they contain reference types, that make encoding much less straightforward).